sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs

```diff
@@ -118,5 +118,10 @@ async ValueTask<OperationState> IOperation.UpdateStateAsync(bool async, Cancella
                     return OperationState.Failure(response, ex);
             }
         }
+
+        /// <summary>
+        /// To get the Id of the operation.
+        /// </summary>
+        public string GetOperationId() => Id;
```

Comment:
We already expose this. What advantage is there of doing it two different ways, potentially causing confusion? These LROs - the whole of Key Vault packages - are HLCs.

Reply:
Yeah, I noticed this doe not make sense for data plane SDKs as the LRO implementation class usually both inherits `Operation<T>` and implements `IOperation`, making `Operation.Id` and `IOperation.GetOperationId()` duplicate on the same class.

This was originally added for management plane as its LRO implementation class only inherits `Operation<T>` and it has a private `_operation` property to do real work which eventually relies on an `IOperation` class. Therefore, it only has `Operation.Id` exposed and relies on the internal `IOperation.GetOperationId()`.

I will check with Michael how we can improve it.

Reply:
Could we make the relatively newer management plane operations look like data plane and try to unify the APIs a bit e.g., an `Id` property instead of a `GetOperationId()` method? /cc @m-nash

Reply:
@heaths this is all prototyping so nothing is set in stone as of yet.  The issue with a property of `ID` is that the value can change at any moment during the execution.

The user scenario as I understand it is as follows:

- Start an LRO and retrieve an instance to an Operation
- save the ID from that LRO instance somewhere
- later construct an Operation passing in an instance to a client an the ID saved from above
- Check status / wait for completely / save and rehydrate again at a later date

The value of ID can change with each iteration of the poller and definitely changes as you cross from 3 high level states `Inital`, `Polling`, `Final`.  This is leading us towards having a method GetOperationId instead of a property as this doesn't give off a false impression that the ID is a known static value that doesn't change for a given instance of Operation.